File: app/src/main/java/me/patrick/aikodigital/pontocerto/network/ServiceApi.java
package me.patrick.aikodigital.pontocerto.network;

import java.util.List;

import me.patrick.aikodigital.pontocerto.model.response.BusPositionResponse;
import me.patrick.aikodigital.pontocerto.model.response.BusStopResponse;
import me.patrick.aikodigital.pontocerto.model.BusStopPositionModel;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Query;

public interface ServiceApi {

    @GET("/Parada/Buscar?termosBusca=")
    Call<List<BusStopPositionModel>> getStopPositions();

    @GET("/Previsao/Parada")
    Call<BusStopResponse> getStopVehicles(@Query("codigoParada") int code);

    @GET("/Posicao/Linha")
    Call<BusPositionResponse> getBusPositions(@Query("codigoLinha") int codigoLinha);

}

File: app/app/src/main/java/com/example/app/data/api/ServiceAPI.kt
package com.example.app.data.api

import com.example.app.data.model.AllLinesResponse
import com.example.app.data.model.LineResponse
import com.example.app.data.model.StopPointResponse
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

interface ServiceAPI {
    @GET("Posicao")
    suspend fun getLines(): AllLinesResponse

    @GET("Posicao/Linha")
    suspend fun getLineByCode(
        @Query("codigoLinha") code: String
    ): LineResponse

    @GET("Parada/Buscar")
    suspend fun getStopPointByLine(
        @Query("termosBusca") code: String
    ): List<StopPointResponse>

    @POST("Login/Autenticar")
    suspend fun authenticate(
        @Query("token") token: String
    ): Response<Boolean>
}
